src/back/spv/writer.rs

```diff
@@ -1772,8 +1786,28 @@ impl Writer {
             .to_words(&mut self.logical_layout.ext_inst_imports);
 
         if self.flags.contains(WriterFlags::DEBUG) {
-            self.debugs
-                .push(Instruction::source(spirv::SourceLanguage::GLSL, 450));
+            if let Some(file_name) = (|| {
+                Path::new(ir_module.file_path.as_ref()?)
+                    .file_name()?
+                    .to_str()
+            })() {
+                let source_file_id = self.id_gen.next();
+                self.source_file_id = Some(source_file_id);
+                self.debugs
+                    .push(Instruction::string(file_name, source_file_id));
+            }
+
+            let debug_source_code = (|| {
+                Some((
+                    self.source_file_id?,
+                    ir_module.source_code.as_ref()?.as_str(),
+                ))
+            })();
+            self.debugs.push(Instruction::source(
+                spirv::SourceLanguage::GLSL,
+                450,
```

Comment:
```suggestion
                spirv::SourceLanguage::Unknown,
                0,
```
Maybe we should use `Unknown` here instead.